<launch>
  <node pkg="map_server" type="map_server" name="map_server" args="/home/spark/test.yaml"/>
  <include file="$(find nx_description)/launch/bigi_description.launch"/>
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 1 map odom 100"/>
  <node pkg="tf" type="static_transform_publisher" name="odom_basefootprint" args="0 0 0 0 0 0 1 odom base_footprint 100"/>
  <include file="$(find b_orbcamera)/launch/b_orbcamera_nodelet.launch"/>

<!--此两个代价地图用于move_base规划路径-->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <param name="shutdown_costmaps" value="false" />

    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="3.0" />

    <param name="oscillation_timeout" value="5.0" />
    <param name="oscillation_distance" value="0.5" />

	<param name="planner_frequency" value="1.0" />
	<param name="planner_patience" value="5.0" />

	<param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />
	<param name="base_global_planner" value="navfn/NavfnROS" />
	<!--
	<param name="base_global_planner" value="navfn/NavfnROS" />
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
	-->

	<param name="recovery_behavior_enabled" value="true" />
	<param name="clearing_rotation_allowed" value="true" />

	<param name="conservative_reset_dist" value="2.0" />

    <rosparam file="$(find pkg_test_movebase)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find pkg_test_movebase)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find pkg_test_movebase)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find pkg_test_movebase)/param/global_costmap_params.yaml" command="load" />

	<rosparam file="$(find pkg_test_movebase)/param/navfn_global_planner_params.yaml" command="load" />
	<rosparam file="$(find pkg_test_movebase)/param/base_local_planner_params.yaml" command="load" />
	<!--
    <rosparam file="$(find spark_rtabmap_movebase)/param/dwa_local_planner_params.yaml" command="load" />
  <remap from="/cmd_vel" to="velocity_smoother/raw_cmd_vel"/>
	-->
    <!--remap from="/cmd_vel" to="/velocity_smoother/raw_cmd_vel"/-->
    <remap from="/odom" to="/encoder_odom" />
  </node>

<!--此全局代价地图用于查看是否可以make plan-->
  <node pkg="pkg_test_movebase" type="test_movebase_node" name="test_movebase_node" output="screen">
    <rosparam file="$(find pkg_test_movebase)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find pkg_test_movebase)/param/global_costmap_params.yaml" command="load" />
  </node>
</launch>
